compiler/include/baseAST.h

```diff
@@ -288,6 +288,9 @@ static inline bool isSymbol(const BaseAST* a)
 static inline bool isType(const BaseAST* a)
 { return a && isType(a->astTag); }
 
+static inline bool isLcnSymbol(const BaseAST* a)
```

Comment:
I think it would be better to just compare against the RTT of the object here, as in:
   return dynamic_cast<LcnSymbol*>(a);

We should deprecate the the astTag field and associated enumeration.  Given that the same information can be obtained from the RTT of the object, it's just dumb to carry that around.

For the purposes of this patch, however, it is acceptable to add a comment labelling this as a TODO, and move on.